custom_components/opnsense/__init__.py

```
@@ -465,268 +438,187 @@ async def _migrate_2_to_3(hass: HomeAssistant, config_entry: ConfigEntry) -> boo
-                )
-            except ValueError as e:
+                _LOGGER.debug("[migrate_2_to_3] new_main_dev: %s", new_dev)
+            except dr.DeviceIdentifierCollisionError as e:
                 _LOGGER.error(
-                    "Error migrating entity: %s. %s: %s",
-                    ent.entity_id,
+                    "Error migrating device: %s. %s: %s",
+                    dev.identifiers,
                     type(e).__name__,
                     e,
                 )
```

Ruff (0.15.18)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)